testSettings.py

```diff
@@ -37,9 +35,42 @@ def _post_check(self) -> None:
         self._lock: threading.Lock
         object.__setattr__(self, "_lock", threading.Lock())
 
-        # Check that the cfg_descr has a connection/test_case_id
+        # Access some properties here. We would get an exception early on, if
+        # there is something wrong.
         self.connection
-        self.test_case_id
+        self.test_case_id.info
+        self.conf_server
+        self.conf_client
+
+    @property
+    def conf_server(self) -> testConfig.ConfServer:
+        # For now, only one server/client is supported and TestConfig already
+        # enforces that. Do tuple-unpacking here, to further assert that there
+        # is only one server/client.
+        (c_server,) = self.connection.server
+        return c_server
+
+    @property
+    def conf_client(self) -> testConfig.ConfClient:
+        # For now, only one server/client is supported and TestConfig already
+        # enforces that. Do tuple-unpacking here, to further assert that there
+        # is only one server/client.
+        (c_client,) = self.connection.client
+        return c_client
+
+    @property
+    def conf_server_used(self) -> testConfig.ConfBaseClientServer:
```

Comment:
What does `conf_server_used` function name supposed to mean?

Would having NodeName in the function name help with reading? Also it is based on the test case, maybe making that explicit would help people understand the function.

Reply:
it's the effectively used `ConfClient` or `ConfServer`, depending on the `TestCaseType`'s `is_same_node`

Reply:
How about ServerConfByTestCaseType?